framework/src/governance/lib/datazone/datazone-custom-asset-type-factory.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CustomResource, Duration, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { CfnProjectMembership } from 'aws-cdk-lib/aws-datazone';
+import { Effect, IRole, ManagedPolicy, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { ILogGroup } from 'aws-cdk-lib/aws-logs';
+import { Construct } from 'constructs';
+import { DataZoneCustomAssetTypeProps } from './datazone-custom-asset-type-props';
+import { DataZoneHelpers } from './datazone-helpers';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+import { DsfProvider } from '../../../utils/lib/dsf-provider';
+
```

Comment:
Should we support the bring your own token approach?

Reply:
We don't need because: 1/ there will not be a huge number of custom assets; 2/ custom asset types should be deployed centrally and there is one stack maximum per asset